### PR TITLE
[SPARK-49017][SQL][TESTS][FOLLOW-UP] Add testing for CreateTableAs with parameters

### PR DESCRIPTION
**Pull request by mihailomilosevic2001**
### What changes were proposed in this pull request?
Addition of a test for `CreateTableAsSelect`.

### Why are the changes needed?
While looking through parameter marker support in statements it was noticed that CacheTableAsSelect actually should not work with parameter markers, but returns UNBOUND_SQL_PARAMETER. (https://issues.apache.org/jira/browse/SPARK-49398 ticket which should solve this issue). Because of this, I suggest adding additional test for `CreateTableAsSelect` to make sure all paths work as expected.

### Does this PR introduce _any_ user-facing change?
No, only tests added.

### How was this patch tested?
N/A

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Review by MaxGekk — COMMENTED** [review]
@mihailom-db Please, add the `[TESTS]` tag, and make PR's title more precise like "Check parameters in CREATE TABLE .. AS"

File: sql/core/src/test/scala/org/apache/spark/sql/ParametersSuite.scala [hunk 1]
```diff
@@ -682,6 +682,24 @@ class ParametersSuite extends QueryTest with SharedSparkSession with PlanTest {
     }
   }
 
+  test("SPARK-49017: bind named parameters with IDENTIFIER clause in create table as") {
+    withTable("testtab", "testtab1") {
+      // Create table
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/ParametersSuite.scala:687]: It is useless comment, I think. Let's remove it.

File: sql/core/src/test/scala/org/apache/spark/sql/ParametersSuite.scala [hunk 2]
```diff
@@ -682,6 +682,24 @@ class ParametersSuite extends QueryTest with SharedSparkSession with PlanTest {
...
+      spark.sql("create table testtab (id int, name string)")
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/ParametersSuite.scala:688]:
```suggestion
      sql("create table testtab (id int, name string)")
```

File: sql/core/src/test/scala/org/apache/spark/sql/ParametersSuite.scala [hunk 3]
```diff
@@ -682,6 +682,24 @@ class ParametersSuite extends QueryTest with SharedSparkSession with PlanTest {
...
+
+      // Insert into table using single param
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/ParametersSuite.scala:690]: hmm, where is the single param?

File: sql/core/src/test/scala/org/apache/spark/sql/ParametersSuite.scala [hunk 4]
```diff
@@ -682,6 +682,24 @@ class ParametersSuite extends QueryTest with SharedSparkSession with PlanTest {
...
+      sql("insert into testtab values(1, 'test1')")
+
+      // cache table with parameters in query
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/ParametersSuite.scala:693]: Sorry, I didn't get where is the cache here.

File: sql/core/src/test/scala/org/apache/spark/sql/ParametersSuite.scala [hunk 5]
```diff
@@ -682,6 +682,24 @@ class ParametersSuite extends QueryTest with SharedSparkSession with PlanTest {
...
+      spark.sql(
```
[the remark this answers is not in the document]
Comment (mihailom-db) [reply sql/core/src/test/scala/org/apache/spark/sql/ParametersSuite.scala:692]: Not possible, compiler complains when we use only sql with parameters.

**Comment by mihailom-db** [comment]
@MaxGekk ready for merge

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @mihailom-db.

**Pull request closed without merging**